sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -46,6 +47,16 @@ object OrcUtils extends Logging {
     "ZLIB" -> ".zlib",
     "LZO" -> ".lzo")
 
+  def rawSize(hadoopConf: Configuration, filePath: Path): BigInt = {
+    val fs = filePath.getFileSystem(hadoopConf)
+    val readerOptions = OrcFile.readerOptions(hadoopConf).filesystem(fs)
+    try {
+      Utils.tryWithResource(OrcFile.createReader(filePath, readerOptions))(_.getRawDataSize)
+    } catch {
+      case _: IOException => BigInt(0)
```

Comment:
Could you add a function comment about this `0`? I'm wondering if this is safely used from the caller side.

Reply:
I can change the result type from `BigInt` to `Option[BigInt]`. This way a possible access problem of the underlaying file well covered.